run.py

```diff
@@ -1,9 +1,25 @@
-from app import cli, create_app, db
+from app import cli, create_app, db, search_client, index
 from app.models import Category, Language, Resource, db
+import os
 
 app = create_app()
 cli.register(app, db)
 
+with app.app_context():
+    if os.environ.get('INDEX_SEARCHED') == "FALSE":
+        query = Resource.query
+        indicies = search_client.list_indices()
+        for ind in indicies['items']:
+            if ind['name'] == os.environ.get('INDEX_NAME'):
+                curr_index = ind
+
+                if curr_index['entries'] != query.count():
```

Comment:
This feels a little brittle still. Do we have to reindex the whole thing if this count is wrong? Also, what if a bunch of resources changed somehow, but it turned out to be the same count? I'm not really sure I know a better way to do it though... @kylemh Do you have any suggestions? Or do you happen to know if this is a best practice?

Reply:
One option is so iterate over every resource and see if it exists in the index and all of its properties are correct. If not we can add it to a batch of operations to update the index. It would take longer than the current solution but would cut down on our operations. 

They have a browse method to return our whole index. https://www.algolia.com/doc/api-reference/api-methods/browse/

Reply:
This needs to not be in the way of the user experience. 

We don't want this to occur in the main application logic. I don't recall if every request has to wait for this but it's in run.py which we forward the requests to. 

How often does this section of code need to be ran?

It looks like this only needs to be done when we are out of sync from algolia and our db. Maybe this only needs to occur on the update resource (and in a background celery task) or even only on every deploy.

Reply:
I don't think it even needs to be run on _every_ deploy. Do you think it needs to be run that often @Digbigpig ?